cloud/cloud_env_impl.cc

```diff
@@ -2007,6 +2009,10 @@ Status CloudEnvImpl::RollNewCookie(const std::string& local_dbname,
   auto newCloudManifest = cloud_manifest_->clone();
   Status st;
   std::string old_epoch = newCloudManifest->GetCurrentEpoch();
+  if (!newCloudManifest->AddEpoch(delta.file_num, delta.epoch)) {
+    return Status::InvalidArgument("Delta already applied in cloud manifest");
+  }
```

Comment:
Returning `InvalidArgument` is okay here because we can say it's invalid to call `RollNewCookie` with an old epoch.